src/main/java/org/symphonyoss/symphony/messageml/elements/Password.java

```diff
@@ -0,0 +1,154 @@
+        setAttribute(NAME_ATTR, getStringAttribute(item));
+        break;
+      case REQUIRED_ATTR:
+        setAttribute(REQUIRED_ATTR, getStringAttribute(item));
+        break;
+      case PLACEHOLDER_ATTR:
+        setAttribute(PLACEHOLDER_ATTR, getStringAttribute(item));
+        break;
+      case MINLENGTH_ATTR:
+        setAttribute(MINLENGTH_ATTR, getStringAttribute(item));
+        break;
+      case MAXLENGTH_ATTR:
+        setAttribute(MAXLENGTH_ATTR, getStringAttribute(item));
+        break;
+      default:
+        throw new InvalidInputException("Attribute \"" + item.getNodeName()
+            + "\" is not allowed in \"" + getMessageMLTag() + "\"");
+    }
+  }
+
+  @Override
+  public void asPresentationML(XmlPrintStream out) {
+    Map<String, String> presentationAttrs = buildTextFieldInputAttributes();
+    out.printElement(PRESENTATIONML_INPUT_TAG, presentationAttrs);
+  }
+
+  @Override
+  public org.commonmark.node.Node asMarkdown() {
+    if (getAttribute(PLACEHOLDER_ATTR) != null) {
+      return new FormElementNode(MARKDOWN, ":" + getAttribute(PLACEHOLDER_ATTR));
+    } else {
+      return new FormElementNode(MARKDOWN, "");
+    }
+  }
+
+  private Map<String, String> buildTextFieldInputAttributes() {
+    Map<String, String> presentationAttrs = new LinkedHashMap<>();
+
+    presentationAttrs.put(PRESENTATIONML_TYPE_ATTR, PRESENTATIONML_INPUT_TYPE);
+    presentationAttrs.put(NAME_ATTR, getAttribute(NAME_ATTR));
+
+    if (getAttribute(PLACEHOLDER_ATTR) != null) {
+      presentationAttrs.put(PLACEHOLDER_ATTR, getAttribute(PLACEHOLDER_ATTR));
+    }
+
+    if (getAttribute(REQUIRED_ATTR) != null) {
+      presentationAttrs.put(REQUIRED_ATTR, getAttribute(REQUIRED_ATTR));
+    }
+
+    if (getAttribute(MINLENGTH_ATTR) != null) {
+      presentationAttrs.put(MINLENGTH_ATTR, getAttribute(MINLENGTH_ATTR));
+    }
+
+    if (getAttribute(MAXLENGTH_ATTR) != null) {
+      presentationAttrs.put(MAXLENGTH_ATTR, getAttribute(MAXLENGTH_ATTR));
+    }
+
+    return presentationAttrs;
+  }
+
+  private void validateMinAndMaxLengths() throws InvalidInputException {
+    int maxLength = 0;
+    int minLength = 0;
+
+    if (getAttribute(MAXLENGTH_ATTR) != null) {
+      maxLength = getLengthAttributeAsInt(MAXLENGTH_ATTR);
+    }
+
+    if (getAttribute(MINLENGTH_ATTR) != null) {
+      minLength = getLengthAttributeAsInt(MINLENGTH_ATTR);
+    }
+
+    if (maxLength > 0 && (minLength > maxLength)) {
+      throw new InvalidInputException("The attribute \"minlength\" must be lower than the \"maxlength\" attribute");
+    }
+  }
+
+  private int getLengthAttributeAsInt(String attributeName) throws InvalidInputException {
+    String errorMessage = String.format("The attribute \"" + attributeName + "\" must be between %s and %s", MIN_ALLOWED_LENGTH,
+        MAX_ALLOWED_LENGTH);
+
+    try {
+      int value = Integer.parseInt(getAttribute(attributeName));
+      if (value < MIN_ALLOWED_LENGTH || value > MAX_ALLOWED_LENGTH) {
```

Comment:
I don't think this validation should be on this method as the method implies only getting the attribute as an integer. Do the length validation only on the "validateMinAndMaxLengths" method.